gslib/commands/rsync.py

```diff
@@ -1460,24 +1464,18 @@ def _RsyncFunc(cls, diff_to_apply, thread_state=None):
       if dst_url.IsCloudUrl():
         dst_url = StorageUrlFromString(diff_to_apply.dst_url_str)
         dst_generation = GenerationFromUrlAndString(dst_url, dst_url.generation)
-        dst_obj_metadata = gsutil_api.GetObjectMetadata(
-            dst_url.bucket_name,
-            dst_url.object_name,
-            generation=dst_generation,
-            provider=dst_url.scheme,
-            fields=['acl'])
-        if dst_obj_metadata.acl:
-          # We have ownership, and can patch the object.
+        try:
+          # Assume we have permission, and can patch the object.
           gsutil_api.PatchObjectMetadata(dst_url.bucket_name,
                                          dst_url.object_name,
                                          obj_metadata,
                                          provider=dst_url.scheme,
                                          generation=dst_url.generation)
-        else:
-          # We don't have object ownership, so it must be copied.
+        except:
```

Comment:
Blanket exceptions can be risky. Looks like we should expect this to raise ServiceException so we can try catching that and add the exception message in debug log. https://github.com/GoogleCloudPlatform/gsutil/blob/11b9f1f4cc160180b85227ccf1df40b0797dab0e/gslib/cloud_api.py#L418
```suggestion
        except ServiceException as err:
          cls.logger.debug('Error while trying to patch: %s', err)
```